client/src/client/__init__.py

```diff
@@ -256,10 +256,23 @@ async def _handle_downstream_system_traffic(self) -> None:
                     payload, message.payload["peer"]
                 )
             elif message.header.msg_id == CansMsgId.ACK_MESSAGE_DELIVERED:
-                self.db_manager.update_message(
-                    id=message.payload["cookie"],
-                    state=CansMessageState.DELIVERED,
+                original_message = self.db_manager.get_message(
+                    id=message.payload["cookie"]
                 )
+                if original_message is not None:
```

Comment:
Please add an "else" clause and log a warning, e.g. "Received unexpected ACK for message {...} from {...}"

Reply:
I deleted this, because the sender works so my overcomplicating isn't needed anymore

Reply:
Added requested changes